tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+  - *                      # remove all paths
+```
+
+
+
+## Misc
+
+Add tool specific controls as literal strings that are directly passed to the individual tools.
+
+YML structure:
+```yml
+misc:                      # Start a list of literal control strings that are directly passed to the tools.
+  - compiler:              # select the toolchain that the literal control string applies too (AC6, IAR, GCC).
+    C: string              # applies to *.c files only.
+    CPP: string            # applies to *.cpp files only.
+    C*: string             # applies to *.c and *.cpp files.
+    ASM: string            # applies to assembler source files
+    Link: string           # applies to the linker
+    Lib: string            # applies to the library manager or archiver
+```
+
+
+## Solution: Collection of related Projects
+
+The section [Project setup for related projects](#project-setup-for-related-projects) describes the collection of related projects.  The file `*.csolution.yml` describes the relationship of this projects.  This file may also define [Target and Build Types](#target-and-build-types) before the section `solution:`.
+
+The YML structure of the section `solution:` is:
+
+```yml
+solution:                  # Start a list of projects.
+  - project:               # path to the project file (required).
+    for-type:              # include project for a list of *build* and *target* types (optional).
+    not-for-type:          # exclude project for a list of *build* and *target* types (optional).
+    compiler:              # specify a specific compiler
+    optimize:              # optimize level for code generation (optional)
+    debug:                 # generation of debug information (optional)
+    defines:               # define symbol settings for code generation (optional).
+    undefines:             # remove define symbol settings for code generation (optional).
+    add-paths:             # additional include file paths (optional).
+    del-paths:             # remove specific include file paths (optional). 
+    misc:                  # Literal tool-specific controls
+```
+
+## Project Definition
+
+Start of a project definition in a `*.cproject.yml` file (optional section)
+
+The YML structure of the section `project:` is:
+```yml
+project:                     # do we need this section at all, perhaps for platform?
+  compiler: name             # specify compiler (AC6, GCC, IAR)
+  description:               # project description (optional)
+  output-type: lib | exe     # generate executable (default) or library
+  processor:                 # specify processor attributes 
+  groups:                    # source files organized in groups
+  layers:                    # software layers of pre-configured software components
+  components:                # software components
+```  
+
+
+## Processor Attributes
+
+The `processor:` keyword defines attributes such as TrustZone and FPU register usage.
```

Comment:
I'm not clear when this processor is supposed to be used ?
Is it for instance to create libraries that could run on any cortex-M4 ?